src/Processors/QueryPlan/SortingStep.cpp

```diff
@@ -88,6 +89,18 @@ SortingStep::SortingStep(
     output_stream->sort_mode = DataStream::SortMode::Stream;
 }
 
+void SortingStep::updateInputStream(DataStream input_stream)
+{
+    input_streams.clear();
```

Comment:
Sort description may be invalid with a new input_stream if there were positions in there.

Actually, I don't know in which case positions are used. Maybe we can remove this code and see what will be broken:
https://github.com/ClickHouse/ClickHouse/blob/31c367d3cd3aefd316778601ff6565119fe36682/src/Core/SortDescription.h#L49-L55

Reply:
let's see https://github.com/ClickHouse/ClickHouse/pull/35805